joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstCreator.scala

```diff
@@ -1408,19 +1414,30 @@ class AstCreator(filename: String, phpAst: PhpFile, fileContent: Option[String],
   private def astForArrayExpr(expr: PhpArrayExpr): Ast = {
     val idxTracker = new ArrayIndexTracker
 
-    val tmpIdentifier = getTmpIdentifier(expr, Some(TypeConstants.Array))
+    val tmpName = getNewTmpName()
+
+    def newTmpIdentifier: Ast = Ast(identifierNode(expr, tmpName, s"$$$tmpName", TypeConstants.Array))
+
+    val tmpIdentifierAssignNode = {
+      val emptyArrayCall = callAst(newOperatorCallNode(PhpOperators.emptyArray, s"[]", line = line(expr)))
+
+      val assignCode = s"$$$tmpName = ${emptyArrayCall.rootCodeOrEmpty}"
+      val assignNode = newOperatorCallNode(Operators.assignment, assignCode, line = line(expr))
+      callAst(assignNode, newTmpIdentifier :: emptyArrayCall :: Nil)
+    }
 
     val itemAssignments = expr.items.flatMap {
-      case Some(item) => Option(assignForArrayItem(item, tmpIdentifier.name, idxTracker))
+      case Some(item) => Option(assignForArrayItem(item, tmpName, idxTracker))
       case None =>
         idxTracker.next // Skip an index
         None
     }
     val arrayBlock = blockNode(expr)
 
     Ast(arrayBlock)
```

Comment:
Every identifier needs a LOCAL variable to track that they refer to the same point in memory. So you would want to create a local variable for the tmp identifier and link REF edges from the identifier to the local. e.g.
```scala
Ast(/* the block */)
 .withChild(Ast(localNode))
 .withRefEdge(localNode, identifierNode)
```

Reply:
php2cpg uses a separate step to create and link locals (`LocalCreationPass`) - so they shouldn't be added during AST creation.